# 
# Submitted-By: Marc Kleine-Budde, 2005-01-31
# Committed-By: Marc Kleine-Budde
#
# Error: 
#
# No error
#
# Description: 
#
# adds the manual method to ifupdown.
#
# State:
#
# This is a improved version of
# http://www.busybox.net/lists/busybox/2006-January/017937.html
#
Index: networking/ifupdown.c
===================================================================
--- a/networking/ifupdown.c.orig
+++ b/networking/ifupdown.c
@@ -567,6 +567,11 @@ static int wvdial_down(struct interface_
 				"-p /var/run/wvdial.%iface% -s 2", ifd, exec));
 }
 
+static int manual_up_down(struct interface_defn_t *ifd, execfn *exec)
+{
+	return(1);
+}
+
 static struct method_t methods[] =
 {
 	{ "wvdial", wvdial_up, wvdial_down, },
@@ -575,6 +580,7 @@ static struct method_t methods[] =
 	{ "bootp", bootp_up, static_down, },
 	{ "dhcp", dhcp_up, dhcp_down, },
 	{ "loopback", loopback_up, loopback_down, },
+	{ "manual", manual_up_down, manual_up_down, },
 };
 
 static struct address_family_t addr_inet =

